Framework/Algorithms/src/PolarizationCorrections/DepolarizedAnalyserTransmission.cpp

```diff
@@ -0,0 +1,215 @@
+  auto const &outputGroup = std::string(PropNames::GROUP_OUTPUT);
+  setPropertyGroup(std::string(PropNames::OUTPUT_WORKSPACE), outputGroup);
+  setPropertyGroup(std::string(PropNames::OUTPUT_FIT), outputGroup);
+  setPropertyGroup(std::string(PropNames::OUTPUT_COV_MATRIX), outputGroup);
+}
+
+std::map<std::string, std::string> DepolarizedAnalyserTransmission::validateInputs() {
+  std::map<std::string, std::string> result;
+  MatrixWorkspace_sptr const &depWs = getProperty(std::string(PropNames::DEP_WORKSPACE));
+  if (depWs->getNumberHistograms() != 1) {
+    result[std::string(PropNames::DEP_WORKSPACE)] =
+        "The depolarised workspace must contain a single spectrum. Contains " +
+        std::to_string(depWs->getNumberHistograms()) + " spectra.";
+  }
+  MatrixWorkspace_sptr const &mtWs = getProperty(std::string(PropNames::MT_WORKSPACE));
+  if (mtWs->getNumberHistograms() != 1) {
+    result[std::string(PropNames::MT_WORKSPACE)] =
+        "The empty cell workspace must contain a single spectrum. Contains " +
+        std::to_string(mtWs->getNumberHistograms()) + " spectra.";
+  }
+
+  if (!WorkspaceHelpers::matchingBins(*depWs, *mtWs, true)) {
+    result[std::string(PropNames::DEP_WORKSPACE)] =
+        "The bins in the DepolarisedWorkspace and EmptyCellWorkspace do not match.";
+  }
+  return result;
+}
+
+void DepolarizedAnalyserTransmission::exec() {
+  auto const &dividedWs = calcDepolarisedProportion();
+  calcWavelengthDependentTransmission(dividedWs, getPropertyValue(std::string(PropNames::OUTPUT_WORKSPACE)));
+}
+
+MatrixWorkspace_sptr DepolarizedAnalyserTransmission::calcDepolarisedProportion() {
+  MatrixWorkspace_sptr const &depWs = getProperty(std::string(PropNames::DEP_WORKSPACE));
+  MatrixWorkspace_sptr const &mtWs = getProperty(std::string(PropNames::MT_WORKSPACE));
+  auto divideAlg = createChildAlgorithm("Divide");
+  divideAlg->setProperty("LHSWorkspace", depWs);
+  divideAlg->setProperty("RHSWorkspace", mtWs);
+  divideAlg->execute();
+  return divideAlg->getProperty(std::string(PropNames::OUTPUT_WORKSPACE));
+}
+
+void DepolarizedAnalyserTransmission::calcWavelengthDependentTransmission(MatrixWorkspace_sptr const &inputWs,
+                                                                          std::string const &outputWsName) {
+  auto funcStream = FitValues::createFunctionStrStream();
+  funcStream << "," << FitValues::EMPTY_CELL_TRANS_NAME << "="
+             << getPropertyValue(std::string(PropNames::EMPTY_CELL_TRANS_START));
+  funcStream << "," << FitValues::DEPOL_OPACITY_NAME << "="
+             << getPropertyValue(std::string(PropNames::DEPOL_OPACITY_START));
+  auto const &func = FunctionFactory::Instance().createInitialized(funcStream.str());
+  auto fitAlg = createChildAlgorithm("Fit");
+  fitAlg->setProperty("Function", func);
+  fitAlg->setProperty("InputWorkspace", inputWs);
+  fitAlg->setProperty("IgnoreInvalidData", true);
+  fitAlg->setProperty("StartX", FitValues::START_X);
```

Comment:
This is taken from Diego's scripts and since it's in wavelength it should be a fairly solid known where the fit should take place. I'll check to make sure they don't want a parameter for this though.